static/js/search/search_input.tsx

```diff
@@ -0,0 +1,285 @@
+        )}
+      </div>
+      <span
+        className="material-icons search-input-search-button"
+        onClick={() => redirectAction(inputText, "")}
+      >
+        search
+      </span>
+    </>
+  );
+
+  function onInputChanged(input: string): void {
+    setInputText(input);
+    if (input === "") {
+      setShowResults(false);
+      setResults({ placeResults: [], svResults: [] });
+      return;
+    }
+    setHoveredIdx(0);
+    setShowResults(true);
+    clearTimeout(delayTimer.current);
+    delayTimer.current = setTimeout(() => search(input), DELAY_MS);
+  }
+
+  function search(query: string): void {
+    latestQuery.current = query;
+    if (placeAutocompleteService.current) {
+      placeAutocompleteService.current.getPredictions(
+        { input: query, types: ["(regions)"] },
+        (predictions, status) =>
+          onAutocompleteCompleted(query, predictions, status)
+      );
+    } else {
+      getSvResultsPromise(query)
+        .then((results) => {
+          if (query !== latestQuery.current) {
+            return;
+          }
+          setResults({ placeResults: [], svResults: results });
+        })
+        .catch(() => {
+          if (query !== latestQuery.current) {
+            return;
+          }
+          setShowResults(false);
+          setResults({ placeResults: [], svResults: [] });
+        });
+    }
+  }
+
+  function onAutocompleteCompleted(
+    query: string,
+    predictions: google.maps.places.AutocompletePrediction[],
+    status: google.maps.places.PlacesServiceStatus
+  ): void {
+    let namedPlacePromise: Promise<NamedPlace[]> = Promise.resolve([]);
+    if (status === google.maps.places.PlacesServiceStatus.OK) {
+      const placeIds = predictions.map((prediction) => prediction.place_id);
+      namedPlacePromise = getPlaceDcids(placeIds).then((dcids) => {
+        const placeDcids = Object.values(dcids);
+        return getPlaceNames(placeDcids).then((names) => {
+          return Object.keys(names)
+            .map((dcid) => {
+              return {
+                dcid,
+                name: names[dcid],
+              };
+            })
+            .filter((place) => !_.isEmpty(place));
+        });
+      });
+    }
+    Promise.all([namedPlacePromise, getSvResultsPromise(query)])
+      .then(([placeResults, svResults]) => {
+        if (query !== latestQuery.current) {
+          return;
+        }
+        if (_.isEmpty(placeResults) && _.isEmpty(svResults)) {
+          setShowResults(false);
+          setResults({ placeResults: [], svResults: [] });
+        } else {
+          setResults({ placeResults, svResults });
+        }
+      })
+      .catch(() => {
+        if (query !== latestQuery.current) {
+          return;
+        }
+        setShowResults(false);
+        setResults({ placeResults: [], svResults: [] });
+      });
+  }
+
+  function handleKeydownEvent(
+    event: React.KeyboardEvent<HTMLDivElement>
+  ): void {
+    if (showResults) {
```

Comment:
maybe do an early return here if !showResults to simplify this handler

Reply:
done